### ISSUE TITLE
[GraphQL]  Internal server error is returned for giftRegistryTypeSearch

### ISSUE THREAD

**Issue by magento-engcom-team** [magento/magento2#31430]
#### Pre-Requisites:

Create few gift registries for a customer.
 search for giftregistry for the above customer
#### Steps:

Run the below query:
```json
{customer
{
  giftRegistryTypeSearch
  (
    firstName:"d" 
    lastName:"thampy" 
    typeUid:1
  )
  {
    event_date
    event_title
    name
    type
    gift*registry*uid
    location
    
  }
}
}
{code}
#### Expected Result:

List of all the gift registries for the customer is returned
#### Actual Result

Internal server error.
 !giftRegistryTypeSearch-internal server error.png|thumbnail!

Error log shows this exception:
{code:java}
report.ERROR: Argument 1 passed to Magento\Framework\GraphQl\Query\Uid::decode() must be of the type string, int given, called in /app/vendor/magento/module-gift-registry-graph-ql/Model/Resolver/SearchType.php on line 57 {"exception":"<object] (GraphQL\\Error\\Error(code: 0): Argument 1 passed to Magento\\Framework\\GraphQl\\Query\\Uid::decode() must be of the type string, int given, called in /app/vendor/magento/module-gift-registry-graph-ql/Model/Resolver/SearchType.php on line 57 at /app/vendor/webonyx/graphql-php/src/Error/Error.php:174, TypeError(code: 0): Argument 1 passed to Magento\\Framework\\GraphQl\\Query\\Uid::decode() must be of the type string, int given, called in /app/vendor/magento/module-gift-registry-graph-ql/Model/Resolver/SearchType.php on line 57 at /app/vendor/magento/framework/GraphQl/Query/Uid.php:26)"} []
```
#### Please note that the argument typeUid for giftRegistryTypeSearch should be of type ID ( not Int) -> That needs to be fixed as well in the scope of this bug fix

Cloud instance where the issue can be reproduced:[https://24develop-2hpekvq-t6ea5z2gu335w.us-4.magentosite.cloud/>

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Oscar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-41265` were created

**Issue Available**: @engcom-Oscar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by cpartica** [comment]
fixed in 2.4.3

**Comment by magento-engcom-team** [comment]
Hi @magento-engcom-team, @engcom-Kilo. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-41265` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
